development/webpack/utils/plugins/SelfInjectPlugin/index.ts

```diff
@@ -142,6 +142,7 @@ export class SelfInjectPlugin {
       `\`\\n//# sourceURL=\${${this.options.sourceUrlExpression(file)}};\``,
     );
     newSource.add(`;`);
+    newSource.add(`s.nonce=btoa(browser.runtime.getURL('/'));`);
```

Comment:
I think the `SelfInjectPlugin`'s options should be extended to accept a `nonceExpression` option, similar to its existing `sourceURLExpression` option, that _defaults_ to the string `"(globalThis.browser||chrome).runtime.getURL('/')"`.

So this line here would become something like:

```typescript
newSource.add(`s.nonce=${this.options.nonceExpression(file)}`);
```